realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/mongodb/sync/SyncedRealmTests.kt

```diff
@@ -508,6 +509,107 @@ class SyncedRealmTests {
+        val expectedObjectId = ObjectId()
+        val expectedUUID = UUID.randomUUID()
+        var expectedRealmObject = SyncDog().apply {
+            id = expectedObjectId
+        }
+        val expectedRealmList = RealmList<SyncDog>()
+        val expectedStringList = RealmList<String>("hello world 1", "hello world 2")
+        val expectedBinaryList = RealmList<ByteArray>()
+        val expectedBooleanList = RealmList<Boolean>(true, false, false, true)
+        val expectedLongList = RealmList<Long>(0, 1, 2, 5, 7)
+        val expectedDoubleList = RealmList<Double>(0.0, 2.toDouble(), 10.5)
+        val expectedDateList = RealmList<Date>(Date(100), Date(10), Date(200))
+        val expectedDecimal128List = RealmList<Decimal128>(Decimal128(10), Decimal128(100), Decimal128(20))
+        val expectedObjectIdList = RealmList<ObjectId>(ObjectId(Date(1000)), ObjectId(Date(100)), ObjectId(Date(2000)))
+        val expectedUUIDList = RealmList<UUID>(UUID.randomUUID())
+
+        Realm.getInstance(config1).use { realm ->
+            realm.executeTransaction {
+                expectedRealmObject = realm.copyToRealmOrUpdate(expectedRealmObject)
+                expectedRealmList.add(expectedRealmObject)
+
+                realm.createObject(SyncAllTypes::class.java, primaryKeyValue).apply {
+                    columnString = expectedString
+                    columnLong = expectedLong
+                    columnDouble = expectedDouble
+                    isColumnBoolean = expectedBoolean
+                    columnDate = expectedDate
+                    columnBinary = expectedBinary
+                    columnDecimal128 = expectedDecimal128
+                    columnObjectId = expectedObjectId
+                    columnUUID = expectedUUID
+                    columnRealmInteger.set(expectedRealmInteger)
+                    columnRealmObject = expectedRealmObject
+                    columnRealmList = expectedRealmList
+                    columnStringList = expectedStringList
+                    columnBinaryList = expectedBinaryList
+                    columnBooleanList = expectedBooleanList
+                    columnLongList = expectedLongList
+                    columnDoubleList = expectedDoubleList
+                    columnDateList = expectedDateList
+                    columnDecimal128List = expectedDecimal128List
+                    columnObjectIdList = expectedObjectIdList
+                    columnUUIDList = expectedUUIDList
+                }
+            }
+            realm.syncSession.uploadAllLocalChanges()
+            assertEquals(1, realm.where<SyncAllTypes>().count())
+        }
+
+        val user2: User = createNewUser()
+        val config2: SyncConfiguration = createDefaultConfig(user2, partitionValue)
+        Realm.getInstance(config2).use { realm ->
+            assertEquals(0, realm.where<SyncAllTypes>().count())
+
+            realm.syncSession.downloadAllServerChanges(5, TimeUnit.SECONDS).let {
```

Comment:
5 seconds feel dangerously low. We have some constants in TestHelper I would probably use instead. I think the lowest is something like 100 seconds, which is arguably quite a lot, but having to rerun tests due to spurious timeouts are super annoying.